### PR TITLE
Precondition setters

### PR DESCRIPTION
**Pull request by carolinaWolox**
## Summary
- Preconditions setters
## Trello Card
https://trello.com/c/iVme2Qbr/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -70,6 +73,7 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
+        Preconditions.checkNotNull(genre, "genre cant be null");
```
Comment (julianvasquez-wolox) [inline src/main/java/wolox/training/models/Book.java:77]: Let's add some extra conditions, like not to be an empty string. :D

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -118,6 +131,8 @@ public String getYear() {
     }
 
     public void setYear(String year) {
+
```
Comment (julianvasquez-wolox) [inline src/main/java/wolox/training/models/Book.java:141]: You can add an extra condition here, the year should be a positive number and be less than the actual year :D

**Review by julianvasquez-wolox — COMMENTED** [review]
Check the comments

**Review by julianvasquez-wolox — APPROVED** [review]
Good job 👍 🥳

**Review by juanwongramirez — APPROVED** [review]
Great!

**Pull request merged**